### PR TITLE
chore: fix pre-commit failures

### PR DESCRIPTION
**Pull request by m4tx**
Pre-commit hooks used to fail on the project's master branch when
running on all files; this commit fixes that.

### CODE REVIEW

**Comment by ConsoleRunner** [comment]
It seems that the PR only contains re-formatting changes, none of which are connected to the pre-commit *config* (you might be missing the config file changes). Additionally, while running all formatters in root might look like a good idea, we intend to place frontend (non-python) files in a `/frontend` file beside `/backend`, which may cause problems with some autoformatters down the road.

Additionally, it seems that `pyproject.toml` was omitted due to the fact that the migrations were not excluded from being reformatted. Omitting some files from autoformatting was our decision, and we're happy to discuss it in more detail, but it looks like the respective config was ignored.

**Comment by m4tx** [comment]
> In general, we try to avoid reformatting the migrations files. Presumably, you ran `pre-commit run --all-files`, which in turn executed `black <path of every file in the project>` under the hood, so black basically disregarded the configurations from `pyproject.toml`. One way to fix it is providing `--force-exclude` options as done in [here](https://github.com/psf/black/issues/1985#issuecomment-778664880). Please, do it in this PR.

No, it doesn't solve the issue at all. You don't have to run `pre-commit run -a` to observe this issue - you can just add an empty comment in `backend/fregepoc/indexers/migrations/0004_add_sourceforgeindexer.py` (or just any other file I've modified in this PR) to either face auto-reformatting or linter issues. If you don't want black to run on certain files, exclude them in pre-commit config; not in pyproject.toml (which is ignored by `pre-commit` command, and therefore, hooks anyway).

> What about reformatted migrations, I do not have anything against those particular changes, as those files seem to have already been reformatted once, but in the future, we would like to omit such diffs.

Again, exclude them from being automatically reformatted in pre-commit hooks.

> What exactly did it fix? It seems that the PR only contains re-formatting changes, none of which are connected to the pre-commit _config_ (you might be missing the config file changes).

I haven't changed the pre-commit config - and that's the whole point of this PR. I've indeed run `pre-commit run -a`, and I expected everything to be okay - but it wasn't.

> Additionally, while running all formatters in root might look like a good idea, we intend to place frontend (non-python) files in a `/frontend` file beside `/backend`, which may cause problems with some autoformatters down the road.

It *will* happen unless you exclude them from being run in `.pre-commit-config.yaml`.

> Moreover, it seems that `pyproject.toml` was omitted (migrations were not excluded from being reformatted). Omitting some files from autoformatting was our decision, and we're happy to discuss it in more detail, but it shows that the respective config was ignored (possibly due to `--all-files` flag).

I guess my comments above answer that as well.

**Comment by apardyl** [comment]
+1 to what @m4tx reported

**Comment by m4tx** [comment]
Anyway, it wasn't really the point of this PR, but I've modified the pre-commit config. Next time, please make sure that the files in your repository conform to the `pre-commit` hooks *before* you push them forcefully to everyone.

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -26,13 +26,16 @@ repos:
             'flake8-no-pep420==2.0.0',
         ]
         args: ["--config=.flake8"]
```
Comment (vladfedoriuk) [inline .pre-commit-config.yaml:28]: changing this one to `  args: ["--config=backend/.flake8"]` should fix file exclusion problem, and consequently there will be no need to duplicate the regex.

File: .pre-commit-config.yaml [hunk 2]
```diff
@@ -26,13 +26,16 @@ repos:
...
+        exclude: '/(migrations|\.direnv|\.eggs|\.git|\.hg|\.mypy_cache|\.nox|\.tox|\.venv|venv|\.env|env|\.svn|_build|buck-out|build|dist)/'
   - repo: https://github.com/PyCQA/isort
     rev: 5.10.1
     hooks:
       - id: isort
         args: ["--settings-file=backend/pyproject.toml"]
+        exclude: '/(migrations|\.direnv|\.eggs|\.git|\.hg|\.mypy_cache|\.nox|\.tox|\.venv|venv|\.env|env|\.svn|_build|buck-out|build|dist)/'
```
Comment (vladfedoriuk) [inline .pre-commit-config.yaml:35]: `isort` doesn't really need this when profile is set to `black`.

File: .pre-commit-config.yaml [hunk 3]
```diff
@@ -26,13 +26,16 @@ repos:
...
   - repo: https://github.com/ambv/black
     rev: 22.3.0
     hooks:
       - id: black
         args: ["--config=backend/pyproject.toml"]
+        exclude: '/(migrations|\.direnv|\.eggs|\.git|\.hg|\.mypy_cache|\.nox|\.tox|\.venv|venv|\.env|env|\.svn|_build|buck-out|build|dist)/'
```
Comment (vladfedoriuk) [inline .pre-commit-config.yaml:41]: Did you try changing `exclude` option in  `[tool.black]` section of `pyproject.toml` to `force-exclude` as I mentioned in a comment earlier?

Comment (vladfedoriuk) [reply .pre-commit-config.yaml:41]: If it works out with `force-exclude`, this regex will also turn out to be redundant.

**Comment by vladfedoriuk** [comment]
> Anyway, it wasn't really the point of this PR, but I've modified the pre-commit config. Next time, please make sure that the files in your repository conform to the `pre-commit` hooks _before_ you push them forcefully to everyone.

Thanks a lot for a kind reminder, though I do not quite understand the ` push them forcefully` part as `pre-commit` config was merged in the same way as any other branch.

File: .pre-commit-config.yaml (hunk 3 above)
Comment (m4tx) [reply .pre-commit-config.yaml:41]: Yes, it seems it works with `force-exclude`. I've updated the PR.

**Comment by m4tx** [comment]
> > Anyway, it wasn't really the point of this PR, but I've modified the pre-commit config. Next time, please make sure that the files in your repository conform to the `pre-commit` hooks _before_ you push them forcefully to everyone.
> 
> Thanks a lot for a kind reminder, though I do not quite understand the ` push them forcefully` part as `pre-commit` config was merged in the same way as any other branch.

What I've meant is that, because `.pre-commit-config.yaml` exists in the repository, all of the contributors are supposed to use it. It's kind of unfortunate for them to run into linter issues when modifying some files on the repo just because nobody thought to run the checks on all files before pushing the config. Note that `pre-commit` runs the checks on the entire file that was modified, not just the modified lines - so it's not too difficult to eventually face the problems because of that.

**Review by apardyl — APPROVED** [review]
LGTM, pls rebase

**Review by kkiermasz — APPROVED** [review]

**Review by ConsoleRunner — CHANGES_REQUESTED** [review]
Rebase requested

**Comment by m4tx** [comment]
@ConsoleRunner done

**Pull request merged**